content/docs/03-ai-sdk-core/15-tools-and-tool-calling.mdx

```diff
@@ -693,15 +693,34 @@ const mcpClient = await createMCPClient({
 
 #### Closing the MCP Client
 
-After initialization, always close the MCP client when you're done to prevent resource leaks. Use try/finally or cleanup functions in your framework:
+After initialization, always close the MCP client after the response is finished to prevent resource leaks. 
+
+When streaming responses, close the client when the LLM response has finished. For example, when using `streamText`, you can use the `onFinish` callback:
```

Comment:
"can" -> "should" (and maybe add an explanation - e.g. "so the connection is closed after any tool calls")